server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/code/JdbcAuthorizationCodeServices.java

```diff
@@ -0,0 +1,81 @@
+package org.cloudfoundry.identity.uaa.oauth.provider.code;
+
+import org.cloudfoundry.identity.uaa.oauth.provider.OAuth2Authentication;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.support.SqlLobValue;
+import org.cloudfoundry.identity.uaa.oauth.common.util.SerializationUtils;
+import org.springframework.util.Assert;
+
+import javax.sql.DataSource;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+
+/**
+ * Moved class implementation of from spring-security-oauth2 into UAA
+ *
+ * The class was taken over from the legacy project with minor refactorings
+ * based on sonar.
+ *
+ * Scope: OAuth2 server
+ */
+public class JdbcAuthorizationCodeServices extends RandomValueAuthorizationCodeServices {
+
+	private static final String DEFAULT_SELECT_STATEMENT = "select code, authentication from oauth_code where code = ?";
+	private static final String DEFAULT_INSERT_STATEMENT = "insert into oauth_code (code, authentication) values (?, ?)";
+	private static final String DEFAULT_DELETE_STATEMENT = "delete from oauth_code where code = ?";
```

Comment:
Interesting that the library dictates the table and it is not a parameter with oauth_code as a default.

Reply:
yes and this is one reason why I said we should move the code to our code base because UAA has a strong dependency also to the data model of this library. Same was with oauth_client_details.